vhost-device-sound/src/device.rs

```diff
@@ -155,86 +154,58 @@ impl VhostUserSoundThread {
         let mut any = false;
 
         for desc_chain in requests {
-            let descriptors: Vec<_> = desc_chain.clone().collect();
-            if descriptors.len() < 2 {
-                return Err(Error::UnexpectedDescriptorCount(descriptors.len()).into());
-            }
+            let mem = atomic_mem.memory();
 
-            // Request descriptor.
-            let desc_request = descriptors[0];
-            if desc_request.is_write_only() {
-                return Err(Error::UnexpectedWriteOnlyDescriptor(0).into());
-            }
+            let mut reader = desc_chain
+                .clone()
+                .reader(&mem)
+                .map_err(|_| Error::DescriptorReadFailed)?;
 
-            let request = desc_chain
-                .memory()
-                .read_obj::<VirtioSoundHeader>(desc_request.addr())
+            let mut writter_status = desc_chain
```

Comment:
just some small typo: 
```suggestion
            let mut writer_status = desc_chain
```

Reply:
Thanks! this typo is in another places too. I'll fix them.